packages/dev/core/src/Cameras/camera.ts

```diff
@@ -819,8 +819,8 @@ export class Camera extends Node {
             return this._computedViewMatrix;
         }
 
-        this.updateCache();
         this._computedViewMatrix = this._getViewMatrix();
+        this.updateCache();

```

Comment:
@Popov72 and @sebavan, I'm tagging you on this line because I just want to make sure that this change makes sense.  With the original tests that checked onViewMatrixChangedObservable, they pass because there were only changes to the x and y rotation params (not z) so the upVector wasn't updated for them.  Since we're checking all axes, there are more opportunities to update the upVector.  Because we update the cache before updating the upVector (done as a part of `_getViewMatrix()` in TargetCamera.  There is a good chance that we cache the new rotation values and update the view matrix and upVector.  This means that we're not catching the upVector change as a part of the first getViewMatrix call.  If `scene.render()` is called, getViewMatrix will run twice (once during the first camera update call, updating just the rotation before caching and then a second time, inside of processSubCameras which will still run because of the difference in upVector values).  This potentially doubles the work from a fresh getViewMatrix call.  If we reverse the order of the highlighted lines to cache AFTER calling `_getViewMatrix`, we will make all changes and then update the cache.  This _should_ ensure a proper view matrix update and cache all changes.  Because I'm leaving for a bit, I wanted to put in this explanation to see if this solution would work.

Reply:
I do not think you can change this one here as easily the cache and synchronization code is unfortunately running really deep within the camera code. I ll flag as draft until you are back but this needs probably way more digging ?

Reply:
Also was it not already an issue before ?